bathbot/src/commands/osu/relax/profile.rs

```diff
@@ -0,0 +1,351 @@
+};
+
+pub(super) async fn relax_profile(orig: CommandOrigin<'_>, args: RelaxProfile<'_>) -> Result<()> {
+    let msg_owner = orig.user_id()?;
+    let config = match Context::user_config().with_osu_id(msg_owner).await {
+        Ok(config) => config,
+        Err(err) => {
+            let _ = orig.error(GENERAL_ISSUE).await;
+
+            return Err(err);
+        }
+    };
+
+    let user_id = match user_id!(orig, args) {
+        Some(user_id) => user_id,
+        None => match config.osu {
+            Some(user_id) => UserId::Id(user_id),
+            None => return require_link(&orig).await,
+        },
+    };
+
+    let user_args = UserArgs::rosu_id(&user_id, GameMode::Osu).await;
+
+    let user = match Context::redis().osu_user(user_args).await {
+        Ok(user) => user,
+        Err(UserArgsError::Osu(OsuError::NotFound)) => {
+            let content = match user_id {
+                UserId::Id(user_id) => format!("User with id {user_id} was not found"),
+                UserId::Name(name) => format!("User `{name}` was not found"),
+            };
+
+            return orig.error(content).await;
+        }
+        Err(err) => {
+            let _ = orig.error(GENERAL_ISSUE).await;
+            let err = Report::new(err).wrap_err("Failed to get user");
+
+            return Err(err);
+        }
+    };
+
+    let user_id = user.user_id.to_native();
+    let client = Context::client();
+    let info_fut = client.get_relax_player(user_id);
+
+    let guild = orig.guild_id();
+    let user_id_fut = Context::user_config().discord_from_osu_id(user_id);
+
+    let (info_res, user_id_res) = tokio::join!(info_fut, user_id_fut);
+
+    let discord_id = match user_id_res {
+        Ok(user) => match (guild, user) {
+            (Some(guild), Some(user)) => Context::cache()
+                .member(guild, user) // make sure the user is in the guild
+                .await?
+                .map(|_| user),
+            _ => None,
+        },
+        Err(err) => {
+            warn!(?err, "Failed to get discord id from osu user id");
+
+            None
+        }
+    };
+
+    let info_res = match info_res {
+        Ok(Some(info_res)) => info_res,
+        Ok(None) => {
+            return orig
+                .error(format!("User `{}` not found", user.username))
+                .await;
+        }
+        Err(err) => {
+            warn!(?err, "Failed to fetch user relax player info");
+
+            let _ = orig.error(GENERAL_ISSUE).await;
+
+            return Err(err);
+        }
+    };
+
+    let origin = MessageOrigin::new(orig.guild_id(), orig.channel_id());
+    let pagination = RelaxProfileArgs::new(user, discord_id, info_res, origin);
+
+    let graph = match relax_playcount_graph(&pagination) {
+        Ok(graph) => graph,
+        Err(err) => {
+            warn!(?err, "Failed to draw a relax playcount graph");
+
+            let _ = orig.error(GENERAL_ISSUE).await;
+
+            return Err(err);
```

Comment:
Instead of logging a warning here you should `err.wrap_err("...")` because the returned error will be logged too.